marshal.go

```diff
@@ -242,6 +243,7 @@ sendRetry:
 
 		packetOut.RequestID = reqID
 
+	resendSameAttempt:
```

Comment:
I'm not a fan of this. If we need another control loop around this section, I would prefer that over a goto jump.

Reply:
I agree. I wasn't fond of it either, but opted to match the existing style. Happy to refactor if you prefer.